### DISCUSSION TITLE
srsgNB GPIO Support

### DISCUSSION THREAD

**Discussion by jboateng1** [srsran/srsRAN_Project#43] [General Help]
Hello folks,

I am trying to run srsRAN with TDD Rf frontEnds (PA and LNA). I wonder if there is a portion of the code that can be changed to support these frontEnds? In OAI, this functionality is in the path "/openairinterface5g/targets/ARCH/USRP/USERSPACE/LIB/usrp_lib.cpp". Thanks for your help.

**Comment by rainey19** [comment]
I'm assuming you're using a USRP. You can look in the file _lib/radio/uhd)/radio_uhd_tx_stream.cpp_ in the function _radio_uhd_tx_stream::transmit_block_, and in the if block _if (metadata.start_of_burst)_ add you're code, then for switching back do something similar in _radio_uhd_rx_stream.cpp_.

Let me know if it works!

**Reply by jboateng1** [reply]
Thank you rainey19 for your response. I have made the changes and compiled successfully. However, I need to test this with our amplifier. I will do so and give you update accordingly.
In the meantime, I am checking if it's possible to get the srsRAN gNB to connect with a USRP based srsUE.

**Reply by pgawlowicz** [reply]
Please check our tutorial on connecting srsUE to gNB: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html

**Comment by joaquim-srs** [comment]
Hi there,

As of last week's release, srsRAN includes a discontinuous transmission mode which may be helpful for using TDD front-ends. At this moment, the feature is experimental, we are still testing it and open to feedback. You can take a look at the [feature announcement discussion](https://github.com/srsran/srsRAN_Project/discussions/367) for more details.
